### ISSUE TITLE
Timers in persisted machines

### ISSUE THREAD

**Issue by jvalkeal** [spring-projects/spring-statemachine#311]
This obviously is a very tricky. Lets say that there is a timer which fires and may take machine to some other state. Nothing will happen if machine is not running(as it is hibernated into storage). Should think some ways to do this.

**Comment by davidtepel** [comment]
We have similar issue - or it is the same one?
We have noticed that the timers are not removed if the machine is removed, i.e creating a new machine instance.
This causes us memroy leaks in the application: 
![image](https://cloud.githubusercontent.com/assets/14327761/23117562/f7daf84c-f758-11e6-8c2f-176f57edc2f8.png)

**Comment by jvalkeal** [comment]
yeah, that's a different issue but I created #321 for it. Tried it and definitely a bug.

**Comment by davidtepel** [comment]
Thanks for the update

**Comment by jvalkeal** [comment]
@davidtepel you could actually check if fix(1.2.3 snapshots) for #321 helped with this issue as I made some fixes to stop triggering when machine is stopped.

**Comment by davidtepel** [comment]
Actually we had to find another solution for the timers becuse of this bug and also, a requirement in which the timeout of the timer should be dynamic. i.e, change the periodic timer timout on the fly without creating a new state machine (in order to change retry mechanism in the same state machine).
So I can't check...
